pkg/build/graph/analysis/bc.go

```diff
@@ -0,0 +1,45 @@
+package analysis
+
+import (
+	"fmt"
+	"strings"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	buildedges "github.com/openshift/origin/pkg/build/graph"
+	buildgraph "github.com/openshift/origin/pkg/build/graph/nodes"
+	imageedges "github.com/openshift/origin/pkg/image/graph"
+	imagegraph "github.com/openshift/origin/pkg/image/graph/nodes"
+)
+
+const (
+	MissingRequiredRegistryWarning = "MissingRequiredRegistry"
+)
+
+// FindUnpushableBuildConfigs checks all build configs that will output to an IST backed by an ImageStream and checks to make sure their builds can push.
+func FindUnpushableBuildConfigs(g osgraph.Graph) []osgraph.Marker {
+	bcNames := []string{}
+
+bc:
+	for _, bcNode := range g.NodesByKind(buildgraph.BuildConfigNodeKind) {
+		for _, istNode := range g.SuccessorNodesByEdgeKind(bcNode, buildedges.BuildOutputEdgeKind) {
+			for _, uncastImageStreamNode := range g.SuccessorNodesByEdgeKind(istNode, imageedges.ReferencedImageStreamGraphEdgeKind) {
+				imageStreamNode := uncastImageStreamNode.(*imagegraph.ImageStreamNode)
+				if len(imageStreamNode.Status.DockerImageRepository) == 0 {
+					bcNames = append(bcNames, bcNode.(*buildgraph.BuildConfigNode).ResourceString())
+					continue bc
+				}
+			}
+		}
+	}
+
+	if len(bcNames) == 0 {
+		return []osgraph.Marker{}
+	}
+
+	return []osgraph.Marker{{
```

Comment:
I'd like discrete markers. If a caller wants to collapse them based on the key and the related nodes, that's always an option. By compressing the markers here you lose information (the actual IST nodes related to the BCs).